app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/SessionsFragment.kt

```diff
@@ -107,30 +172,41 @@ class SessionsViewPagerAdapter(
         fragmentManager: FragmentManager,
         private val activity: Activity
 ) : FragmentStateNullablePagerAdapter(fragmentManager) {
+    companion object {
+        private val startDateFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
+    }
 
     private val fireBaseAnalytics = FirebaseAnalytics.getInstance(activity)
     private var currentTab by Delegates.observable<Tab?>(null) { _, old, new ->
         if (old != new && new != null) {
             val key = when (new) {
                 Tab.All -> AllSessionsFragment::class.java.simpleName
                 is Tab.RoomTab -> RoomSessionsFragment::class.java.simpleName + new.room.name
+                is Tab.TimeTab -> ScheduleSessionsFragment::class.java.simpleName + new.title
             }
             fireBaseAnalytics.setCurrentScreen(activity, null, key)
         }
     }
 
     private val tabs = arrayListOf<Tab>()
     private var roomTabs = mutableListOf<Tab.RoomTab>()
+    private var schedulesTabs = mutableListOf<Tab.TimeTab>()
 
     sealed class Tab(val title: String) {
         object All : Tab("All")
         data class RoomTab(val room: Room) : Tab(room.name)
+        data class TimeTab(val schedule: SessionSchedule) :
+                Tab("Day${schedule.dayNumber} / ${startDateFormat.format(schedule.startTime)}")
     }
 
-    private fun setupTabs() {
+    private fun maySetupTabs(otherTabs: List<Tab>) {
```

Comment:
Hou about using `setupTabIfNeeded` instead of `maySetupTabs`
I think `maySetupTabs` has a random feeling.
What do you think?

Reply:
If you think this is normal, you do not have to change it. 🙇

Reply:
Make sense! Let me rename it. Thanks 👍